holoviews/core/io.py

```diff
@@ -840,3 +841,101 @@ def contents(self, maxlen=70):
     def listing(self):
         "Return a list of filename entries currently in the archive"
         return ['.'.join([f,ext]) if ext else f for (f,ext) in self._files.keys()]
+
+
+
+class ArchiveReader(object):
+    """
+    ArchiveReaders abstract operations over different archive formats,
+    e.g. directory vs. zip file.
+    """
+    def __init__(self,archive):
+        self.archive_location = archive
+        self.root,self.archive_name=os.path.split(self.archive_location)
+
+    def open_member(self,name):
+        raise NotImplementedError
+
+    def member_exists(self,name):
+        raise NotImplementedError
+
+    def list_members(self):
+        raise NotImplementedError
+
+
+
+class Directory(ArchiveReader):
+    
+    def __init__(self,archive):
+        super(Directory,self).__init__(archive)        
+        if not os.path.isdir(self.archive_location):
+            raise ValueError("Directory archive not found at %s"%self.archive_location)
+        self.archive=self.archive_location
+
+    def list_members(self):
+        return os.listdir(self.archive)
+
+    def open_member(self,name):
+        return open(self._fullname(name),'r')
+
+    def member_exists(self,name):
+        return os.path.exists(self._fullname(name))
+
+    def _fullname(self,name):
```

Comment:
I would consider having a ``path`` attribute that points to the directory/archive file. Then you would have:

```python
def open_member(self,name):
        return open(os.path.join(self.path, name),'r')
```